libs/ui/src/screens/token/token.tsx

```diff
@@ -52,14 +56,26 @@ export const Token: FC<Props> = ({ style }) => {
     params: { token }
   } = useRoute<RouteProp<ScreensParamList, ScreensEnum.Token>>();
   const { chainId } = useSelectedNetworkSelector();
+  const dispatch = useDispatch();
 
   const { name, symbol, tokenAddress, decimals, tokenId, thumbnailUri, balance } = token;
+
   const { price, usdPriceChange24h } = useTokenMarketInfoSelector(tokenAddress, chainId);
+  const { tokenBalance, fiatBalance } = useTokenBalance(tokenAddress, tokenId);
 
-  const balanceFromStore = useTokenBalanceSelector(getTokenSlug(tokenAddress, tokenId));
-  const formattedBalance = getFormattedBalance(balanceFromStore ?? balance.data, decimals);
+  const formattedBalance = getFormattedBalance(tokenBalance.toString() ?? balance.data, decimals);
   const isGasToken = checkIsGasToken(tokenAddress);
-  const usdBalance = getDollarValue({ amount: balance?.data ?? 0, price, decimals });
+  const fiatBalanceToDisplay = getFiatBalanceToDisplay(tokenBalance, fiatBalance);
```

Comment:
Same as here https://github.com/madfish-solutions/oko-wallet/pull/255/files#r1087595326  
there will be no need for `getFiatBalanceToDisplay` here and in other places, if you will prepare this data in shared place